### ISSUE TITLE
rule: no constant conditions

### ISSUE THREAD

**Issue by michaelficarra** [eslint/eslint#621]
This rule should report when a literal (`Literal`, `FunctionExpression`, `ObjectExpression`, etc.) is used in the condition position of `ConditionalExpression`s, `IfStatement`s, `WhileStatement`s, etc.

This is likely an error in production code. Example of programs that should be flagged:

``` js
if(true) doSomething();
```

``` js
var a = 0 ? b : c;
```
